### PR TITLE
add doctests for bart like seq2seq models

### PR DESCRIPTION
**Pull request by patil-suraj**
# What does this PR do?

Enable doctests for bart like seq2seq models.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

File: src/transformers/file_utils.py [hunk 1]
```diff
@@ -1012,6 +1012,9 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
     >>> from transformers import {processor_class}, {model_class}
     >>> import torch
 
+    >>> torch.manual_seed(0)  # doctest:+ELLIPSIS
+    <torch._C.Generator object at ...>
```
Comment (sgugger) [inline src/transformers/file_utils.py:1016]:
```suggestion
    >>> torch.manual_seed(0)  # doctest:+IGNORE_OUTPUT
```

File: src/transformers/file_utils.py [hunk 2]
```diff
@@ -1022,42 +1025,65 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
...
 PT_SEQUENCE_CLASSIFICATION_SAMPLE = r"""
     Example of single-label classification:
 
     ```python
-    >>> from transformers import {processor_class}, {model_class}
     >>> import torch
+    >>> from transformers import {processor_class}, {model_class}
+
+    >>> torch.manual_seed(0)  # doctest:+ELLIPSIS
+    <torch._C.Generator object at ...>
```
Comment (sgugger) [inline src/transformers/file_utils.py:1049]:
```suggestion
    >>> torch.manual_seed(0)  # doctest:+IGNORE_OUTPUT
```

File: src/transformers/file_utils.py [hunk 3]
```diff
@@ -1022,42 +1025,65 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
...
 
     Example of multi-label classification:
 
     ```python
-    >>> from transformers import {processor_class}, {model_class}
     >>> import torch
+    >>> from transformers import {processor_class}, {model_class}
+
+    >>> torch.manual_seed(0)  # doctest:+ELLIPSIS
+    <torch._C.Generator object at ...>
```
Comment (sgugger) [inline src/transformers/file_utils.py:1073]:
```suggestion
    >>> torch.manual_seed(0)  # doctest:+IGNORE_OUTPUT
```

**Review by sgugger — COMMENTED** [review]
Ah, GitHub was not showing me everything. ELLIPSIS is activated by default, so there is no need to add it in comments. We do have one that ignores the outputs and should be uses for those torch Generators.

You seem to have one badly closed docstring or something like that.

**Review by patrickvonplaten — APPROVED** [review]
Very nice!

File: src/transformers/file_utils.py [hunk 4]
```diff
@@ -1022,42 +1024,57 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
 
     >>> outputs = model(**inputs, start_positions=start_positions, end_positions=end_positions)
     >>> loss = outputs.loss
+    >>> round(loss.item(), 2)
+    {expected_loss}
```
Comment (patrickvonplaten) [inline src/transformers/file_utils.py:1028]: I went over this PR too quickly :-/ Why didn't we check for the actual answer of the model here? 

The model is a QA model -> it's not very descriptive to just test for the output shape

File: src/transformers/file_utils.py [hunk 5]
```diff
@@ -1022,42 +1024,57 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
...
     >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
-    >>> model = {model_class}.from_pretrained("{checkpoint}")
+    >>> model = {model_class}.from_pretrained("{checkpoint}", num_labels=2)
 
     >>> inputs = tokenizer("Hello, my dog is cute", return_tensors="pt")
     >>> labels = torch.tensor([1]).unsqueeze(0)  # Batch size 1
     >>> outputs = model(**inputs, labels=labels)
     >>> loss = outputs.loss
     >>> logits = outputs.logits
+    >>> list(logits.shape)
```
Comment (patrickvonplaten) [inline src/transformers/file_utils.py:1057]: same here -> we should test that the logits give a sensible answer

File: src/transformers/file_utils.py [hunk 6]
```diff
@@ -1022,42 +1024,57 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
...
 
     >>> tokenizer = {processor_class}.from_pretrained("{checkpoint}")
-    >>> model = {model_class}.from_pretrained("{checkpoint}", problem_type="multi_label_classification")
+    >>> model = {model_class}.from_pretrained("{checkpoint}", problem_type="multi_label_classification", num_labels=2)
 
     >>> inputs = tokenizer("Hello, my dog is cute", return_tensors="pt")
     >>> labels = torch.tensor([[1, 1]], dtype=torch.float)  # need dtype=float for BCEWithLogitsLoss
     >>> outputs = model(**inputs, labels=labels)
     >>> loss = outputs.loss
-    >>> logits = outputs.logits
```
Comment (patrickvonplaten) [inline src/transformers/file_utils.py:1060]: same here
